src/Nethermind/Nethermind.Network/TrustedNodes/TrustedNodesManager.cs

```diff
@@ -0,0 +1,172 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Config;
+using Nethermind.Core.Crypto;
+using Nethermind.Logging;
+using Nethermind.Stats.Model;
+
+namespace Nethermind.Network
+{
+    public class TrustedNodesManager : ITrustedNodesManager
+    {
+        private ConcurrentDictionary<PublicKey, NetworkNode> _nodes = new();
+        private readonly string _trustedNodesPath;
+        private readonly ILogger _logger;
+
+        public TrustedNodesManager(string trustedNodesPath, ILogManager logManager)
+        {
+            _trustedNodesPath = trustedNodesPath.GetApplicationResourcePath();
+            _logger = logManager.GetClassLogger();
+        }
+
+        public IEnumerable<NetworkNode> Nodes => _nodes.Values;
+
+        public async Task InitAsync()
+        {
+            if (!File.Exists(_trustedNodesPath))
+            {
+                if (_logger.IsDebug)
+                {
+                    _logger.Debug($"Trusted nodes file not found at: {_trustedNodesPath}");
+                }
+                return;
+            }
+
+            HashSet<string> lines = new();
+            // Read lines asynchronously
+            await foreach (string line in File.ReadLinesAsync(_trustedNodesPath))
+            {
+                if (!string.IsNullOrWhiteSpace(line))
+                {
+                    lines.Add(line);
+                }
+            }
+
+            if (_logger.IsInfo)
+            {
+                _logger.Info($"Loaded {lines.Count} trusted nodes from: {Path.GetFullPath(_trustedNodesPath)}");
+            }
+
+            if (lines.Any() && _logger.IsDebug)
+            {
+                _logger.Debug("Trusted nodes:\n" + string.Join(Environment.NewLine, lines));
+            }
+
+            // Parse each line into a NetworkNode
+            List<NetworkNode> networkNodes = new();
+            foreach (string line in lines)
+            {
+                try
+                {
+                    networkNodes.Add(new NetworkNode(line));
+                }
+                catch (Exception ex) when (ex is ArgumentException or SocketException)
+                {
+                    if (_logger.IsError)
+                    {
+                        _logger.Error($"Failed to parse '{line}' as a trusted node.", ex);
+                    }
+                }
+            }
+
+            _nodes = new ConcurrentDictionary<PublicKey, NetworkNode>(
+                networkNodes.ToDictionary(n => n.NodeId, n => n));
```

Comment:
You don't need `lines` not `networkNodes` as intermediate steps (you also create `_nodes` 2 times, once in constructor).
Just add to `_nodes` collection directly in one pass.